requestmanager/client.go

```diff
@@ -35,17 +41,30 @@ const (
 	defaultPriority = graphsync.Priority(0)
 )
 
+type state uint64
+
+const (
+	queued state = iota
+	running
+	paused
+)
+
 type inProgressRequestStatus struct {
-	ctx            context.Context
-	startTime      time.Time
-	cancelFn       func()
-	p              peer.ID
-	terminalError  chan error
-	resumeMessages chan []graphsync.ExtensionData
-	pauseMessages  chan struct{}
-	paused         bool
-	lastResponse   atomic.Value
-	onTerminated   []chan<- error
+	ctx              context.Context
+	startTime        time.Time
+	cancelFn         func()
+	p                peer.ID
+	terminalError    error
+	pauseMessages    chan struct{}
+	state            state
+	lastResponse     atomic.Value
+	onTerminated     []chan<- error
+	request          gsmsg.GraphSyncRequest
+	doNotSendCids    *cid.Set
+	nodeStyleChooser traversal.LinkTargetNodePrototypeChooser
+	inProgressChan   chan graphsync.ResponseProgress
+	inProgressErr    chan error
+	traverser        ipldutil.Traverser
```

Comment:
this is a lot of state for a request status.
can none of these be de-duped? e.g. can you not learn the peer.ID from the request?

Reply:
nope unfortuantely you can't.

Thhere are a lot of things to track, but also, I'm totally worried. the block memory trumps this in terms of memory consumption.